internal/transformer/classes/generators/dockerfilegenerators/net5appdockerfilegenerator.go

```diff
@@ -0,0 +1,249 @@
+		err = xml.Unmarshal(byteValue, &configuration)
+		if err != nil {
+			logrus.Errorf("Could not parse the project file %s", err)
+		}
+		serviceName := strings.TrimSuffix(filepath.Base(csprojFile), filepath.Ext(csprojFile))
+		if configuration.PropertyGroup.TargetFramework == dotNETCore5 {
+			namedServices = map[string]transformertypes.ServicePlan{
+				serviceName: []transformertypes.TransformerPlan{{
+					Mode:              t.Config.Spec.Mode,
+					ArtifactTypes:     []transformertypes.ArtifactType{artifacts.ContainerBuildArtifactType},
+					BaseArtifactTypes: []transformertypes.ArtifactType{artifacts.ContainerBuildArtifactType},
+					Paths: map[string][]string{
+						artifacts.ProjectPathPathType: {dir},
+						CsprojFilePathType:            {csprojFile},
+					},
+				}},
+			}
+			return namedServices, nil, nil
+		}
+	}
+	return nil, nil, nil
+}
+
+// Transform transforms the artifacts
+func (t *Net5AppDockerfileGenerator) Transform(newArtifacts []transformertypes.Artifact, oldArtifacts []transformertypes.Artifact) ([]transformertypes.PathMapping, []transformertypes.Artifact, error) {
+	pathMappings := []transformertypes.PathMapping{}
+	artifactsCreated := []transformertypes.Artifact{}
+	for _, a := range newArtifacts {
+		if a.Artifact != artifacts.ServiceArtifactType {
+			continue
+		}
+		relSrcPath, err := filepath.Rel(t.Env.GetEnvironmentSource(), a.Paths[artifacts.ProjectPathPathType][0])
+		if err != nil {
+			logrus.Errorf("Unable to convert source path %s to be relative : %s", a.Paths[artifacts.ProjectPathPathType][0], err)
+		}
+		var sConfig artifacts.ServiceConfig
+		err = a.GetConfig(artifacts.ServiceConfigType, &sConfig)
+		if err != nil {
+			logrus.Errorf("unable to load config for Transformer into %T : %s", sConfig, err)
+			continue
+		}
+		sImageName := artifacts.ImageName{}
+		err = a.GetConfig(artifacts.ImageNameConfigType, &sImageName)
+		if err != nil {
+			logrus.Debugf("unable to load config for Transformer into %T : %s", sImageName, err)
+		}
+		var ports []int32
+		var net5Config Net5TemplateConfig
+		net5Config.AppName = a.Name
+		net5Config.CsprojFilePath, err = filepath.Rel(a.Paths[artifacts.ProjectPathPathType][0], a.Paths[CsprojFilePathType][0])
+		if err != nil {
+			logrus.Errorf("Error while getting the relative path of csproj file%s", err)
+		}
+		jsonFiles, err := common.GetFilesByExt(a.Paths[artifacts.ProjectPathPathType][0], []string{".json"})
+		if err != nil {
+			logrus.Errorf("Error while finding json files %s", err)
+		}
+		for _, jsonFile := range jsonFiles {
+			if filepath.Base(jsonFile) == launchSettingsJSON {
+				launchSettings := LaunchSettings{}
+				if err := common.ReadJSON(jsonFile, &launchSettings); err != nil {
+					logrus.Errorf("unable to read the launchSettings.json file: %s", err)
+					continue
+				}
+				if launchSettings.Profiles[net5Config.AppName] != nil {
+					profiles := launchSettings.Profiles[net5Config.AppName].(map[string]interface{})
```

Comment:
If there is a well defined structure for launchsettings, why not define the structure?

Reply:
Done.